src/Adapter/Order/QueryHandler/GetOrderPreviewHandler.php

```diff
@@ -63,16 +64,24 @@ final class GetOrderPreviewHandler implements GetOrderPreviewHandlerInterface
      */
     private $locale;
 
+    /**
+     * @var int
+     */
+    private $contextLanguageId;
+
     /**
      * @param LocaleRepository $localeRepository
      * @param string $locale
+     * @param int|null $contextLanguageId
      */
     public function __construct(
         LocaleRepository $localeRepository,
-        string $locale
+        string $locale,
+        ?int $contextLanguageId
```

Comment:
If you want to maintain backwards compatibility
```suggestion
        ?int $contextLanguageId = null
```

Reply:
Indeed or you need to mark it in the PR description if the modification is mandatory

However I'm not even sure you need to do this, since you already inject the locale from the context in your service definition `"@=service('prestashop.adapter.legacy.context').getContext().language.getLocale()"` Then you can simply use `Language::getIdByLocale($this->locale)` all the time (performance cost for the extra object is probably negligible)